cmd/lifecycle/creator.go

```diff
@@ -239,6 +241,25 @@ func (c *createCmd) Exec() error {
 		}
 	}
 
+	pingCtx, cancelPing := context.WithCancel(context.Background())
+	defer cancelPing()
+	pingDoneChan := make(chan struct{}, 1)
+	// send pings to docker daemon while BUILDING to prevent connection closure
+	go func() {
+		defer func() { pingDoneChan <- struct{}{} }()
+		if c.docker == nil {
+			return
+		}
+		for {
+			select {
+			case <-time.After(time.Second * 1):
+				_, _ = c.docker.Ping(pingCtx)
+			case <-pingCtx.Done():
+				return
+			}
+		}
+	}()
```

Comment:
Could we wrap this in a named function, to make the code easier to read? I think something like the below would do the trick (I left out the channel as I didn't see the need for it, but maybe I'm missing something):
```
func (c *createCmd) startPing(pingCtx context.Context) {
	if c.docker == nil {
		return
	}

	go func() {
		for {
			select {
			case <-time.After(time.Second * 1):
				_, _ = c.docker.Ping(pingCtx)
			case <-pingCtx.Done():
				return
			}
		}
	}()
}
```